### PR TITLE
Cli: add linesFilter option

### PR DESCRIPTION
**Pull request by yoavf**
Given a json file containing filenames and line numbers, only the strings in these files will be added to the output

Sample format
```
{
	 "client/me/purchases/manage-purchase/index.jsx" :
    [
      "159",
      "160",
      "162",
      "163",
      "164",
      "165",
      "166",
      "167",
      "168",
      "169",
      "175",
      "143"
    ]
}
```

### CODE REVIEW

**Comment by Tug** [comment]
That's cool, what's the use case?

**Comment by yoavf** [comment]
localci-client uses that to only get the strings added in changed/added lines in a diff:
https://github.com/Automattic/gp-localci-client/blob/master/generate-new-strings-pot.sh#L116

**Comment by blowery** [comment]
@yoavf do you still want to land this?

**Comment by yoavf** [comment]
Yes, I do.

File: bin/index.js [hunk 1]
```diff
@@ -67,13 +69,30 @@ inputPaths.forEach( function( inputFile ) {
 	}
 } );
 
+if ( linesFile ) {
+	if ( ! fs.existsSync( linesFile ) ) {
+      console.error( 'Error: linesFile, `' + linesFile + '`, does not exist' );
```
Comment (blowery) [inline bin/index.js]: whitespace is a little off here

File: bin/index.js [hunk 2]
```diff
@@ -67,13 +69,30 @@ inputPaths.forEach( function( inputFile ) {
...
+  }
+
+	lines = JSON.parse( fs.readFileSync( linesFile, 'utf8') );
+	for ( var line in lines ) {
+		lines[line] = lines[line].map( String );
+		var modPath = path.relative( __dirname, line ).replace( /^[\/.]+/, '' );
+		if (  modPath !== line ) {
+			lines[ modPath ] = lines[line];
```
Comment (blowery) [inline bin/index.js]: some inconsistent spacing in here

**Pull request merged**